tests/fixtures/encoder_decoder/nlvr_semantic_parser/experiment.json

```diff
@@ -46,10 +46,14 @@
     "max_decoding_steps": 20,
     "attention_function": {"type": "dot_product"},
     "checklist_cost_weight": 0.8,
+    "dynamic_cost_weight": {
+      "wait_num_epochs": 0,
+      "rate": 0.1
+    },
     "penalize_non_agenda_actions": true
   },
   "iterator": {
-    "type": "bucket",
+    "type": "epoch_tracking_bucket",
```

Comment:
This gives you a high test coverage number, but doesn't actually test anything other than that this doesn't crash.  It'd be nice to actually test this, in two places:

1. A test for the `EpochTrackingBucketIterator`, that makes sure that the epoch gets incremented correctly, for two different datasets.
2. A simple test for your model that checks that the checklist weight parameter is updated like you expect when you get different epoch numbers passed to `forward`.

Reply:
Added two tests. Thanks!